cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -908,6 +911,70 @@ public KafkaBrokerConfigurationBuilder withTieredStorage(String clusterName, Tie
         return this;
     }
 
+    /**
+     * Configures the quotas based on the type of the plugin - {@link QuotasPluginKafka}, {@link QuotasPluginStrimzi}
+     *
+     * @param clusterName    Name of the cluster
+     * @param quotasPlugin   Configuration of the quotas plugin
+     * @return  Returns the builder instance
+     */
+    public KafkaBrokerConfigurationBuilder withQuotas(String clusterName, QuotasPlugin quotasPlugin) {
+        if (quotasPlugin != null) {
+            // for the built-in Kafka quotas plugin we don't need to configure anything
+            if (quotasPlugin instanceof QuotasPluginStrimzi quotasPluginStrimzi) {
+                printSectionHeader("Quotas configuration");
+
+                configureQuotasPluginStrimzi(clusterName, quotasPluginStrimzi);
+
+                writer.println();
+            }
+        }
+
+        return this;
+    }
+
+    /**
+     * Configures Strimzi quotas plugin
+     *
+     * @param clusterName           Name of the cluster
+     * @param quotasPluginStrimzi   Strimzi quotas plugin configuration
+     */
+    private void configureQuotasPluginStrimzi(String clusterName, QuotasPluginStrimzi quotasPluginStrimzi) {
+        // add TopicOperator's and CruiseControl's user to the excluded principals
+        List<String> excludedPrincipals = new ArrayList<>(List.of(
+            String.format("%s-%s", clusterName, "entity-topic-operator"),
+            String.format("%s-%s", clusterName, "cruise-control")
```

Comment:
I was referring to the proposal:

```
Set excluded users for whom quotas will not be applied using the client.quota.callback.static.excluded.principal.name.list option. Regardless of whether custom exclusions are specified, the list of excluded users will automatically include internal users employed by the Topic Operator and Cruise Control. This ensures these internal users are not obstructed, as they might be required for resolving disk space issues:

    The Topic Operator modifies the retention settings of existing topics. Message production and fetching activities are required only for the Bidirectional Topic Operator. Therefore, the Topic Operator user can be removed if support for the Bidirectional Topic Operator is discontinued.
    Cruise Control rebalances topics among brokers to address disk space shortages.

```

but TO can be removed from the list, as BTO was removed.
And for CC -> _I think_ that it should be kept - but corrected to the right value.

Reply:
> I was referring to the proposal

Right, but is it really needed? I'm not sure it is really needed.

Reply:
Does CC produce it? Or the metric reporter running inside Kafka?

Reply:
Also, should the Kafka brokers themself be in the exclude list?

Reply:
> Does CC produce it? Or the metric reporter running inside Kafka?

I'm not CC expert, but I found this:

```
This step is required if CruiseControlMetricsReporter is used for metrics collection (i.e. the default for Cruise Control). The metrics reporter periodically samples the Kafka raw metrics on the broker and sends them to a Kafka topic. 
```
so I think the latter.

> Also, should the Kafka brokers themself be in the exclude list?

TBH I'm not sure as I think that it should be related just to producing messages.


Also regarding this:
```
These are not the identities used by these components, check the authorization configuration for examples of how they should look like
```

I checked the authorization part of the code (from where I added these two, I just picked the CN name here) -> I thought that the Strimzi quotas plugin takes just the user names as `my-user` etc., as it should be a comma-separated list (I was a bit afraid that when I add the `User:CN=my-cluster-cruise-control,O=io.strimzi` for example, it will be wrongly parsed). I will try it.

Reply:
> This step is required if CruiseControlMetricsReporter is used for metrics collection (i.e. the default for Cruise Control). The metrics reporter periodically samples the Kafka raw metrics on the broker and sends them to a Kafka topic. 

But that IMHO doesn't run under the CC identity but under the broker identity.

Reply:
> Does CC produce it? Or the metric reporter running inside Kafka?

Both, the metric reporter produces metrics to Kafka topics, and the CC server consumes those metrics, process them into samples, and then produces those samples to Kafka topics

Reply:
Thanks Kyle! :) I will add both then.

Reply:
I added the identities for CC and brokers (didn't test it yet), but I will convert this PR to draft, as we should wait for the https://github.com/strimzi/kafka-quotas-plugin/issues/49 to be fixed and new version of the quotas plugin to be released.